golden-tests/run-test.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+export OLD_DRV="$(nix-instantiate ./old-derivation/drv.nix)"
```

Comment:
A small nitpick: you probably don't need to `export` them, `old_drv="$(nix-instantiate ./old-derivation/drv.nix)"` will be just enough to make it available in the scope of the remaining script, and AFAICS, these env variables aren't used anywhere in the subsequent calls